### ISSUE TITLE
Site Editing: Saving drafts and switching between template & post

### ISSUE THREAD

**Issue by geheren** [WordPress/gutenberg#27905]
**Describe the bug**
The apply -> save flow had some confusing parts to me. If I make changes to the template, I can only move forward by saving it or getting rid of it. In addition, I am having trouble publishing just a post without publishing the template. Was following the [FSE Outreach Program Testing Call](https://make.wordpress.org/test/2020/12/23/fse-program-testing-call-1-template-editing/).

_This is my first time reporting an issue, so please let me know if I am doing this wrong._

**To reproduce**

1. Install Gutenberg 9.6+ and a FSE ready theme.
2. Go to Posts > Add New
3. Make some changes to the post > Publish or Save Draft
4. Use the option to edit the template to switch into template editing mode in the post settings section.
5. Make some changes to the header, footer, and post content.
6. Select "Apply" and this is where my hangup is. 

**Expected behavior**
To me, I feel like I should be able to save the draft of the changes, as I am used to when using the block editor, without making it go live. I also found the Apply button to be confusing in this flow. 

**Screenshots**
Here is a video walking through my hangups: [https://michaelgeheren.com/fse-issue/](https://michaelgeheren.com/fse-issue/)

**Editor version (please complete the following information):**
- WordPress version: 5.6
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Gutenberg
- If the Gutenberg plugin is installed, which version is it? 9.6.2

**Desktop (please complete the following information):**

- OS: MacOS
- Browser: Safari
- Version:14.0.1 (16610.3.6.3)

**Comment by annezazu** [comment]
Thanks for opening this! I agree and ran into a similar point of confusion: https://github.com/WordPress/gutenberg/issues/27899 For now, I'll leave these both open as I think your issue touches on the idea of saving a draft in a way that the one I opened doesn't.

**Comment by youknowriad** [comment]
I agree as well, it's very confusing and the fact that saving the post and saving the other entities are separate UIs always felt confusing to me. I'd love some design thinking and explorations here before moving forward with code.

**Comment by youknowriad** [comment]
There's a few issues about the saving flow. I think now things are less confusing but still not perfect. There #27896 about a discard button in the save flow for instance. But there's no "apply" button anymore which was the root cause of the confusion.
